src/pkg/clouds/aws/ecs/cfn/template.go

```diff
@@ -46,7 +46,7 @@ func getCacheRepoPrefix(prefix, suffix string) string {
 	return repo
 }
 
-func createTemplate(stack string, containers []types.Container, spot bool) *cloudformation.Template {
+func createTemplate(stack string, containers []types.Container, refs map[string]string, spot bool) *cloudformation.Template {
```

Comment:
`refs` feels like premature generalization, unless you already have a plan for other refs?

Reply:
Agreed, changed to a new type for the override for better type determination, and we can expand the type when needed.